carma-messenger-core/j2735_convertor/src/psm_convertor.cpp

```diff
@@ -0,0 +1,271 @@
+  out_msg.confidence = in_msg.confidence;
+}
+
+void PSMConvertor::convert(const carma_v2x_msgs::msg::AccelerationSet4Way& in_msg, j2735_v2x_msgs::msg::AccelerationSet4Way& out_msg)
+{
+  // Convert Longitudinal and Lateral
+  out_msg.longitudinal = ValueConvertor::valueCavToJ2735<int16_t>(
+      in_msg.longitudinal, units::CM_PER_M, in_msg.presence_vector,
+      carma_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_AVAILABLE, j2735_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_UNAVAILABLE);
+
+  out_msg.lateral = ValueConvertor::valueCavToJ2735<int16_t>(in_msg.lateral, units::CM_PER_M, in_msg.presence_vector,
+                                                             carma_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_AVAILABLE,
+                                                             j2735_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_UNAVAILABLE);
+
+  // Convert Vertical
+  out_msg.vert =
+      ValueConvertor::valueCavToJ2735<int8_t>(in_msg.vert, units::FIFTIETH_G_PER_M_PER_SEC_SQR, in_msg.presence_vector,
+                                              carma_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_VERTICAL_AVAILABLE,
+                                              j2735_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_VERTICAL_UNAVAILABLE);
+
+  // Convert Yaw Rate
+  out_msg.yaw_rate = ValueConvertor::valueCavToJ2735<int16_t>(
+      in_msg.yaw_rate, units::CENTI_DEG_PER_DEG, in_msg.presence_vector,
+      carma_v2x_msgs::msg::AccelerationSet4Way::YAWRATE_AVAILABLE, j2735_v2x_msgs::msg::AccelerationSet4Way::YAWRATE_UNAVAILABLE);
+}
+
+void PSMConvertor::convert(const carma_v2x_msgs::msg::PositionalAccuracy& in_msg, j2735_v2x_msgs::msg::PositionalAccuracy& out_msg)
+{
+  // Convert semi_major Axis
+  out_msg.semi_major = ValueConvertor::valueCavToJ2735<uint8_t>(
+      in_msg.semi_major, units::TWENTIETH_M_PER_M, in_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_AVAILABLE, j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_UNAVAILABLE);
+
+  // Convert semi_minor Axis
+  out_msg.semi_minor = ValueConvertor::valueCavToJ2735<uint8_t>(
+      in_msg.semi_minor, units::TWENTIETH_M_PER_M, in_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_AVAILABLE, j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_UNAVAILABLE);
+
+  // Convert Orientation
+  out_msg.orientation = ValueConvertor::valueCavToJ2735<uint16_t>(
+      in_msg.orientation, units::DEG_360_OVER_65535_PER_DEG, in_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_AVAILABLE,
+      j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_UNAVAILABLE);
+}
+
+void PSMConvertor::convert(const carma_v2x_msgs::msg::PathHistory& in_msg, j2735_v2x_msgs::msg::PathHistory& out_msg)
+{
+    out_msg.curr_gnss_status = in_msg.curr_gnss_status;
+    out_msg.initial_position.lat.latitude = in_msg.initial_position.lat.latitude / latitude_conversion_const_;
```

Comment:
2 things here-
1. Initial_position fields here have a presence_vector value which indicates whether the field was assigned or not.
Before assigning the field we need to check the presence vector. 
2. Since both carma_v2x_msgs and j2735_v2x_msgs versions of this message have a field for presence_vector, we need to set it as well. I guess you can just assign the value from one to the other
out_msg.initial_position.presence_vector = in_msg.initial_position.presence_vector